### PR TITLE
Support including fragment documents inside `gql`

### PR DESCRIPTION
**Pull request by tmeasday**
See https://github.com/apollostack/graphql-fragments/issues/2#issuecomment-258057233 + https://github.com/apollostack/GitHunt-React/tree/fragments-from-graphql-anywhere

### CODE REVIEW

**Comment by tmeasday** [comment]
@stubailo I'm not really sure what the best way to support the second mode. It seems more complicated to figure out that you are mid way through a selection set when you see a fragment document for a substitution

**Comment by tmeasday** [comment]
@stubailo shall we merge and release this?

**Comment by stubailo** [comment]
@tmeasday sorry, I didn't have a chance to review this; some comments:

1. We should test that this preserves the property that queries are === each other if they are the same. That's used in various places to identify whether a query is different or not.
2. There's got to be some way to avoid printing and then parsing again every time. The package has some memoization built in so that we make sure to only parse things once, the first time they are encountered. Maybe there is something similar we can do? Otherwise the fragments will be printed and parsed every single time `gql` runs.

**Comment by tmeasday** [comment]
@stubailo so we are OK with only working with fragment documents that come from this library?

Or are you just saying I shouldn't use this `print` technique and instead figure out how to put the fragment document's AST at the bottom of the query document. That seems pretty reasonable to me, although it might be out of order.

**Comment by stubailo** [comment]
oh, yeah I mean just using the AST directly rather than having the intermediate state of having a string.

**Comment by tmeasday** [comment]
Alright take a look now. As discussed elsewhere, we are keeping the `loc` field on top-level documents so we can avoid an unnecessary `print` here. There may be other alternatives we can use in future.

**Comment by tmeasday** [comment]
@stubailo I think you should probably review this before I merge it.

File: index.js [hunk 1]
```diff
@@ -1,19 +1,22 @@
 var parse = require('./parser').parse;
+var print = require('./printer').print;
```
Comment (stubailo) [inline index.js]: We don't need this import anymore.

File: index.js [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
 
 var cache = {};
 
-function stripLoc (doc) {
+function stripLoc (doc, removeRoot) {
```
Comment (stubailo) [inline index.js]: This argument isn't well named - I think it should either be "isRoot" or "shouldRemoveThisLevel" or something.

Comment (tmeasday) [inline index.js]: I disagree, I think it's the first argument that is poorly named. It should be `ast`. And the second argument is "are we removing the root of the *current ast we are recursing on*".

Comment (tmeasday) [inline index.js]: But I don't care also so I'll happily change it to `isRoot`

Comment (stubailo) [inline index.js]: > are we removing the root of the current ast we are recursing on

If we're going to be pedantic, then it should be called `removeLoc` right?

Comment (tmeasday) [inline index.js]: `removeLocAtThisLevel` it is

**Comment by stubailo** [comment]
Looks great.

**Pull request merged**